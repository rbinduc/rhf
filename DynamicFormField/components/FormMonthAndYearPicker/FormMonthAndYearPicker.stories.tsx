import React from 'react';
import {storiesOf} from '@storybook/react';
import FormMonthAndYearPicker, {IFormMonthAndYearPickerProps} from '.';
import {ThemeProvider} from 'emotion-theming';
import {themeMock} from '../../../../../themes/theme-mock';
import useFormHook from '../../utils/hooks/useFormHook';
import {FormProvider} from 'react-hook-form';

const defaultProps: IFormMonthAndYearPickerProps = {
  id: 'date',
  name: 'date',
  label: 'Date',
  placeholder: 'Select a date',
  defaultValue: 'August, 2021',
};

storiesOf('Organisms/RHF/DynamicForm/FormMonthAndYearPicker', module)
  .addDecorator((getStory) => <ThemeProvider theme={themeMock}>{getStory()}</ThemeProvider>)
  .add('default case', () => {
    const {methods}: any = useFormHook();
    return (
      <FormProvider {...methods}>
        <FormMonthAndYearPicker {...defaultProps} />
      </FormProvider>
    );
  });
