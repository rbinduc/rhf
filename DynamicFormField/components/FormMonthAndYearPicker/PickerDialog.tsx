import React from 'react';
import styled from '@emotion/styled';
import {Dialog, DialogActions, DialogContent} from '@material-ui/core';
import {Typography} from '../../../../../atoms/Typography';
import {renderIcon, RenderIconProps} from '../../../../Header/headerHooks';
import {locators} from '../../../../../../automation/PageLocators';

const {aMonthPickerTitleLabel} = locators.commonLocators;

// ----------------------------------------------------------------------------
// Props
// ----------------------------------------------------------------------------
interface IPickerDialogProps {
  open: boolean;
  onClose: () => void;
  title: string;
  confirmIconProps: RenderIconProps;
  children: React.ReactNode;
}

// ----------------------------------------------------------------------------
// Component
// ----------------------------------------------------------------------------
const PickerDialog = ({open, onClose, title, confirmIconProps, children}: IPickerDialogProps) => (
  <Container open={open} onClose={onClose}>
    <Header>
      <Title variant="h1" id={aMonthPickerTitleLabel}>
        {title}
      </Title>
    </Header>
    <Content>{children}</Content>
    <Footer>{renderIcon(confirmIconProps)}</Footer>
  </Container>
);

// ----------------------------------------------------------------------------
// Styles
// ----------------------------------------------------------------------------
const Container = styled(Dialog)`
  & .MuiDialog-paper {
    width: 20rem;
    border-radius: 10px;
  }
`;

const Title = styled(Typography)`
  font-weight: 600;
  font-size: 1.2rem;
  @media (min-width: 360px) {
    font-size: 1.4rem;
  }
`;

const Header = styled.div`
  display: flex;
  align-items: center;
  height: 75px;
  padding: 8px 24px;
  color: #fff;
  background: linear-gradient(103.18deg, #da291c 0%, #bf0d3e 100%);
`;

const Content = styled(DialogContent)`
  overflow: hidden;
  height: 100%;
`;

const Footer = styled(DialogActions)`
  height: 56px;
  border-top: 1px solid #c7ced6;
  display: flex;
  flex-direction: row;
  padding: 8px 24px;
`;

export default PickerDialog;
