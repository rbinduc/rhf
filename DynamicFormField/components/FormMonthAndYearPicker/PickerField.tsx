import React from 'react';
import styled from '@emotion/styled';
import {TextField, FormControl, FormControlLabel, InputAdornment} from '@material-ui/core';
import {getLabel} from '../../DynamicFormField';
import {IFormTextFieldProps} from '../FormTextField';
import FormMessage from '../../utils/FormMessage';
import {TenantVector} from '../../../../../icons/TenantVector/TenantVector.component';
import {IconNames} from '../../../../../icons/IconNameConstants';

// ----------------------------------------------------------------------------
// Props
// ----------------------------------------------------------------------------
export type IPickerFieldProps = IFormTextFieldProps & {
  isDirty?: boolean;
  errorMessage?: string;
};

// ----------------------------------------------------------------------------
// Component
// ----------------------------------------------------------------------------
const PickerField = ({
  id,
  name,
  label,
  labelId,
  placeholder,
  isRequired,
  isDisabled,
  value,
  isDirty = false,
  errorMessage = '',
  ...rest
}: IPickerFieldProps) => (
  <FormControlStyled error={Boolean(errorMessage)} required={isRequired}>
    <FormControlLabelStyled
      label={getLabel(label, !!isRequired)}
      id={labelId}
      className={!!errorMessage ? 'error' : ''}
      control={
        <TextField
          {...rest}
          id={id}
          name={name}
          placeholder={placeholder}
          value={value}
          autoComplete="off"
          error={Boolean(errorMessage)}
          fullWidth
          variant="outlined"
          helperText={<FormMessage message={errorMessage} component={'span'} />}
          disabled={isDisabled}
          required={isRequired}
          InputProps={{
            readOnly: true,
            className: !isDisabled && isDirty ? 'dirty' : '',
            endAdornment: (
              <InputAdornmentStyled position="start">
                <TenantVector tenantVectorName={IconNames.Event} />
              </InputAdornmentStyled>
            ),
          }}
          InputLabelProps={{
            disableAnimation: true,
            shrink: false,
          }}
        />
      }
    />
  </FormControlStyled>
);

// ----------------------------------------------------------------------------
// Styles
// ----------------------------------------------------------------------------
const FormControlStyled = styled(FormControl)`
  width: 100%;
`;

const FormControlLabelStyled = styled(FormControlLabel)`
  display: flex;
  flex-direction: column-reverse;
  width: 100%;
  align-items: baseline;
  margin: 0;
  & .MuiFormControlLabel-label {
    margin-bottom: 0.3125rem;
  }

  & .MuiInputBase-input {
    text-transform: capitalize;
  }

  & .MuiOutlinedInput-adornedEnd {
    padding-right: 0;
  }
`;

const InputAdornmentStyled = styled(InputAdornment)`
  position: absolute;
  right: 0.5rem;
`;

export default PickerField;
