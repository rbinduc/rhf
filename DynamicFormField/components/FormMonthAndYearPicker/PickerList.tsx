import React from 'react';
import styled from '@emotion/styled';
import {MenuList, MenuItem} from '@material-ui/core';
import {Typography} from '../../../../../atoms/Typography';

// ----------------------------------------------------------------------------
// Props
// ----------------------------------------------------------------------------
interface IPickerListProps<V = any> {
  id?: string;
  options: V[];
  selected: V;
  isDisabled?: (option: any) => boolean;
  handleSelected: (event: React.MouseEvent<HTMLElement>, option: V) => void;
}

// ----------------------------------------------------------------------------
// Component
// ----------------------------------------------------------------------------
const PickerList: React.FC<IPickerListProps> = ({id = '', options = [], selected, isDisabled, handleSelected}: IPickerListProps) => (
  <MenuListStyled>
    {options?.map((option) => (
      <MenuItemStyled
        ref={(ref: any) => {
          if (ref && option === selected) {
            ref?.scrollIntoView({behavior: 'smooth', block: 'center', inline: 'center'});
          }
        }}
        id={`${id}_${option}`}
        key={option}
        disabled={isDisabled && isDisabled(option)}
        selected={option === selected}
        onClick={(event) => handleSelected(event, option)}
        disableRipple
        autoFocus={option === selected}
      >
        <Typography variant="body2">{option}</Typography>
      </MenuItemStyled>
    ))}
  </MenuListStyled>
);

// ----------------------------------------------------------------------------
// Styles
// ----------------------------------------------------------------------------
const MenuListStyled = styled(MenuList)`
  height: 300px;
  overflow-y: auto;
  display: block;
  -ms-overflow-style: none;
  scrollbar-width: none;
  &::-webkit-scrollbar {
    display: none;
  }
  outline: none;
`;

const MenuItemStyled = styled(MenuItem)`
  height: 56px;
  display: flex;
  justify-content: center;
  text-transform: capitalize;
  outline: none;
  &:hover {
    background-color: transparent;
  }
  &.MuiListItem-root.Mui-focusVisible {
    background-color: transparent;
  }
  &.Mui-selected {
    background-color: transparent !important;
  }
  &.Mui-selected p {
    color: #273446;
    font-weight: 700;
    font-size: 21px;
  }
`;

export default PickerList;
