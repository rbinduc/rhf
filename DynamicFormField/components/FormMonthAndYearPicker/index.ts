export {default} from './FormMonthAndYearPicker';
export type {IFormMonthAndYearPickerProps} from './FormMonthAndYearPicker'