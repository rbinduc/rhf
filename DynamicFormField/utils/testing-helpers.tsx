import React from 'react';
import {render} from '@testing-library/react';
import {FormProvider} from 'react-hook-form';
import useFormHook from './hooks/useFormHook';
import {ThemeProvider} from 'emotion-theming';
import {themeMock} from '../../../../themes/theme-mock';

interface IDefaultValue {
  [key: string]: any;
}

interface RenderOptions {
  defaultValues?: IDefaultValue;
  toPassBack?: any[];
}

/**
 * Testing Library utility function to wrap tested component in React Hook Form
 * @param {ReactElement} ui A React component
 * @param objectParameters
 * @param {Object} objectParameters.defaultValues Initial form values to pass into
 * React Hook Form, which you can then assert against
 * @param {Array} objectParameters.toPassBack React Hook Form method names which we'd
 * like to pass back and use in tests.  A primary use case is sending back 'setError',
 * so we can manually setErrors on React Hook Form components and test error handling
 */
export function renderWithDyanmicForm(
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  {defaultValues = {}, toPassBack = []}: RenderOptions = {}
) {
  let dynamicFormMethods = {} as any;

  const Wrapper = ({children}: {children: React.ReactNode}) => {
    const {methods}: any = useFormHook({defaultValues});
    for (let dynamicFormItem of toPassBack) {
      dynamicFormMethods[dynamicFormItem] = methods[dynamicFormItem];
    }

    return (
      <ThemeProvider theme={themeMock}>
        <FormProvider {...methods}>{children}</FormProvider>
      </ThemeProvider>
    );
  };

  return {
    ...render(ui, {wrapper: Wrapper as React.ComponentType<{}>}),
    dynamicFormMethods,
  };
}

/**
 * Higher order helper function which wraps a component w/ React Hook Form
 * @param {React Component} WrappedComponent to pass into
 * @param {*} restProps any other remaining props
 * @returns {React Component}
 */
export function withDynamicForm(WrappedComponent: React.ComponentType<any>, restProps: object, defaultValues: IDefaultValue) {
  return () => {
    const methods = useFormHook({defaultValues});

    return (
      <ThemeProvider theme={themeMock}>
        <WrappedComponent {...restProps} {...methods} />
      </ThemeProvider>
    );
  };
}
