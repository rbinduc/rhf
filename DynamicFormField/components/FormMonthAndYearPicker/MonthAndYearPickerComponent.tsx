import React, {useState} from 'react';
import {useTranslation} from 'react-i18next';
import {Grid} from '@material-ui/core';
import PickerDialog from './PickerDialog';
import PickerList from './PickerList';
import PickerField, {IPickerFieldProps} from './PickerField';
import {IconNames} from '../../../../../icons/IconNameConstants';
import {
  generateYears,
  generateMonths,
  getFormattedMonthAndYear,
  getDateFromMonthAndYear,
  getMonthAndYear,
  IMonthAndYear,
  getMonthNumber,
  getCurrentMonth,
} from '../../../../../../helpers/DateUtils';
import {locators} from '../../../../../../automation/PageLocators';

const {aMonthPickerConfirmButton} = locators.commonLocators;

// ----------------------------------------------------------------------------
// Props
// ----------------------------------------------------------------------------
export type IMonthAndYearPickerComponentProps = IPickerFieldProps & {
  startYear?: number;
  endYear?: number;
  disableFutureDates?: boolean;
};

// ----------------------------------------------------------------------------
// Component
// ----------------------------------------------------------------------------
function useToggle() {
  const [open, setOpen] = useState(false);
  const [touched, setTouched] = useState(false);

  const handleOpen = () => {
    if (!touched) {
      setTouched(true);
    }
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return {open, touched, handleOpen, handleClose};
}

const getIsMonthDisabled = (
  month: string,
  year: string | number,
  selectedMonth: string,
  handleMonth: (event: React.MouseEvent<HTMLElement> | undefined, item: string) => void
): boolean => {
  if (year === new Date().getFullYear()) {
    const currentMonthNumber = getCurrentMonth();
    const monthNumber = getMonthNumber(month);
    const selectedMonthNumber = getMonthNumber(selectedMonth);

    if (selectedMonthNumber > currentMonthNumber) {
      handleMonth(undefined, '');
    }

    return monthNumber > currentMonthNumber;
  }
  return false;
};

const renderPickerList = (
  id: string,
  month: string,
  year: number | '',
  months: string[],
  years: number[],
  disableFutureDates: boolean,
  handleMonth: (event: React.MouseEvent<HTMLElement> | undefined, item: string) => void,
  handleYear: (event: React.MouseEvent<HTMLElement> | undefined, item: number) => void
) => (
  <Grid container>
    <Grid item xs={6}>
      <PickerList
        id={id + 'Month'}
        options={months}
        selected={month}
        handleSelected={handleMonth}
        isDisabled={(option: string) => disableFutureDates && getIsMonthDisabled(option, year, month, handleMonth)}
      />
    </Grid>
    <Grid item xs={6}>
      <PickerList
        id={id + 'Year'}
        options={years}
        selected={year}
        handleSelected={handleYear}
        isDisabled={(option: number) => disableFutureDates && option > new Date().getFullYear()}
      />
    </Grid>
  </Grid>
);

const getMonthAndYearFormatted = (value: string) => getMonthAndYear(getDateFromMonthAndYear(value));

function useMonthAndYear(
  selectedValue: any,
  months: string[],
  years: number[],
  disableFutureDates: boolean,
  handleChange: (date: IMonthAndYear) => void,
  id: string
) {
  const {t} = useTranslation(['common']);
  const {open, touched, handleOpen, handleClose} = useToggle();
  const [{month, year} = getMonthAndYearFormatted(selectedValue), setState] = useState<IMonthAndYear>(
    getMonthAndYearFormatted(selectedValue)
  );
  const handleUpdateMonth = (_event: React.MouseEvent<HTMLElement> | undefined, newMonth: string) =>
    setState((preState) => ({...preState, month: newMonth}));

  const handleUpdateYear = (_event: React.MouseEvent<HTMLElement> | undefined, newYear: number) =>
    setState((preState) => ({...preState, year: newYear}));

  const handleUpdateDate = (date: IMonthAndYear) => handleChange(date);

  const renderPickerDialog = () => (
    <PickerDialog
      title={t('monthPicker.title')}
      open={!!open}
      onClose={handleClose}
      confirmIconProps={{
        id: aMonthPickerConfirmButton,
        icon: true,
        text: t('monthPicker.confirm'),
        name: IconNames.ArrowForward,
        disabled: !month || !year,
        onClick: () => {
          handleUpdateDate({month, year});
          handleClose();
        },
        reverse: true,
      }}
    >
      {renderPickerList(id, month, year, months, years, disableFutureDates, handleUpdateMonth, handleUpdateYear)}
    </PickerDialog>
  );

  return {month, year, renderPickerDialog, touched, handleOpen};
}

const MonthAndYearPickerComponent = React.forwardRef<IMonthAndYearPickerComponentProps, any>((props, ref) => {
  const {
    id = '',
    label,
    labelId,
    name,
    placeholder,
    validations,
    isRequired = false,
    isDisabled = false,
    disableFutureDates = true,
    startYear,
    endYear,
    isDirty = false,
    errorMessage = '',
    ...rest
  } = props;
  const handleChange = async (date: IMonthAndYear) => {
    if (props.value !== getFormattedMonthAndYear(date)) {
      props.onChange(getFormattedMonthAndYear(date));
    }
  };

  const {renderPickerDialog, handleOpen} = useMonthAndYear(
    props.value,
    generateMonths(),
    generateYears(startYear, endYear),
    disableFutureDates,
    handleChange,
    id
  );

  return (
    <>
      <PickerField
        {...{
          ...rest,
          inputRef: ref,
          id,
          label,
          labelId,
          name,
          placeholder,
          validations,
          isRequired,
          isDisabled,
          isDirty,
          errorMessage,
          inputProps: {
            'aria-label': label && typeof label === 'string' ? label : placeholder,
            'aria-hidden': isDisabled,
            ...rest.inputProps,
          },
          onClick: () => {
            !isDisabled && handleOpen();
          },
        }}
      />
      {renderPickerDialog()}
    </>
  );
});

export default MonthAndYearPickerComponent;
