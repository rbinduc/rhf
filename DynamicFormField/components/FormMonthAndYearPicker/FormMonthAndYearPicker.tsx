import React from 'react';
import {Controller, useFormContext} from 'react-hook-form';
import MonthAndYearPickerComponent, {IMonthAndYearPickerComponentProps} from './MonthAndYearPickerComponent';

export type IFormMonthAndYearPickerProps = IMonthAndYearPickerComponentProps;

const FormMonthAndYearPicker = (props: IFormMonthAndYearPickerProps) => {
  const {name, defaultValue = '', validations, ...rest} = props;
  const {
    control,
    formState: {errors, dirtyFields},
  } = useFormContext();
  const errorMessage = errors[name!]?.message;
  const isDirty = !!dirtyFields[name!];

  return (
    <Controller
      control={control}
      defaultValue={defaultValue}
      name={name!}
      rules={validations}
      render={({field}) => <MonthAndYearPickerComponent {...rest} {...field} name={name} errorMessage={errorMessage} isDirty={isDirty} />}
    />
  );
};

export default FormMonthAndYearPicker;
