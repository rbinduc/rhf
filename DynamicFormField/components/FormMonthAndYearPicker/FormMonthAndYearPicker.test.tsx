import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {screen, act} from '@testing-library/react';
import FormMonthAndYearPicker, {IFormMonthAndYearPickerProps} from '.';
import {renderWithDyanmicForm} from '../../utils/testing-helpers';
import {UseFormReturn} from 'react-hook-form';

const defaultProps: IFormMonthAndYearPickerProps = {
  id: 'date',
  name: 'date',
  label: 'Date',
  placeholder: 'Please select a date',
  defaultValue: 'August, 2021',
};

jest.mock('@ionic-native/keyboard', () => ({
  hide: jest.fn(),
}));

jest.mock('../../../../../../automation/PageLocators', () => ({
  locators: {
    jobPageLocators: {
      aFiltersIcon: 'filtersIcon',
    },
    commonLocators: {
      aHeaderText: 'headerText',
    },
    accountSettings: {
      aChangeCountryHeader: 'changeCountryHeader',
    },
  },
}));

jest.mock('../../../../../../helpers/HelperUtils', () => ({
  getTenantIdFromLS: () => 98,
  getI18nFromLS: () => '',
  getNumberFromString: (v: string) => parseInt(v, 10),
}));

describe('when rendering FormMonthAndYearPicker', () => {
  it('snapshot match', () => {
    const {container} = renderWithDyanmicForm(<FormMonthAndYearPicker {...defaultProps} />, {});
    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();

    expect(screen).toMatchSnapshot();
  });

  it('renders in document and a default value and disabled', () => {
    renderWithDyanmicForm(<FormMonthAndYearPicker {...defaultProps} isDisabled={true} />, {
      defaultValues: {
        date: 'August, 2021',
      },
    });
    const dateInput = screen.getByLabelText(defaultProps.label as string);
    expect(screen.getByText('Date')).toBeInTheDocument();
    expect((dateInput as HTMLInputElement).disabled).toBeTruthy();
    expect((dateInput as HTMLInputElement).value).toBeTruthy();
  });

  it('displays error message when errors', async () => {
    const {getByText, dynamicFormMethods} = renderWithDyanmicForm(<FormMonthAndYearPicker {...defaultProps} />, {
      toPassBack: ['setError'],
    });

    await act(async () => {
      (dynamicFormMethods as UseFormReturn)?.setError(defaultProps.name!, {
        type: 'manual',
        message: 'Date is required',
      });
    });

    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(getByText('Date is required'));
  });
});
