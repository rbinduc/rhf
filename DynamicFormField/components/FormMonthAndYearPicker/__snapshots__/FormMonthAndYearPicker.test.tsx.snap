// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering FormMonthAndYearPicker snapshot match 1`] = `
<div
  class="MuiFormControl-root css-b2xexh-FormControlStyled ekdgj6n0"
>
  <label
    class="MuiFormControlLabel-root  css-g9wtmv-FormControlLabelStyled ekdgj6n1"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
      >
        <input
          aria-describedby="date-helper-text"
          aria-hidden="false"
          aria-invalid="false"
          aria-label="Date"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
          id="date"
          name="date"
          placeholder="Please select a date"
          readonly=""
          type="text"
          value="August, 2021"
        />
        <div
          class="MuiInputAdornment-root css-14syd4-InputAdornmentStyled ekdgj6n2 MuiInputAdornment-positionStart"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 14 16"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m12.2383 14.2383v-8.22658h-10.47658v8.22658zm-2.25002-13.500019h1.51172v1.511719h.7383c.3984 0 .75.15234 1.0547.45703s.457.65625.457 1.05469v10.47658c0 .3984-.1523.75-.457 1.0547s-.6563.457-1.0547.457h-10.47658c-.42188 0-.785158-.1523-1.089845-.457-.28125-.3047-.421875-.6563-.421875-1.0547v-10.47658c0-.39844.140625-.75.421875-1.05469.304687-.30469.667965-.45703 1.089845-.45703h.73828v-1.511719h1.51172v1.511719h5.97656zm.77342 8.261719v3.7617h-3.7617v-3.7617z"
              fill="currentColor"
            />
          </svg>
        </div>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-2"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
        id="date-helper-text"
      />
    </div>
    <span
      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      Date
    </span>
  </label>
</div>
`;

exports[`when rendering FormMonthAndYearPicker snapshot match 2`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;
